modules/ai/assets/js/editor/pages/form-layout/index.js

```diff
@@ -86,6 +102,13 @@ const FormLayout = ( { onClose, onInsert, onGeneration, onSelect, DialogHeaderPr
 		lastRun.current();
 
 		setIsPromptEditable( false );
+		setCurrentPage( 1 );
+	};
+
+	const handleRegenerate = () => {
+		setCurrentPage( pagesCount + 1 );
+		regenerate( promptInputRef.current.value );
+		onRegenerate();
```

Comment:
hmm
I probably need an event in the onGenerate actually (right before sending the requests), so I'll be able to clear the previous containers on each new prompt